libjolie/src/jolie/lang/nativeTypes/NativeType.java

```diff
@@ -28,45 +28,80 @@
  *
  * @author Fabrizio Montesi
  */
-public enum NativeType
-{
-	//UNDEFINED( "undefined" ),
-	STRING( "string" ),
-	INT( "int" ),
-	LONG( "long" ),
-	BOOL( "bool" ),
-	DOUBLE( "double" ),
-	VOID( "void" ),
-	RAW( "raw" ),
-	ANY( "any" );
-
-	private final static Map< String, NativeType > idMap = new HashMap<>();
-
-	static {
-		for( NativeType type : NativeType.values() ) {
-			idMap.put( type.id(), type );
-		}
+public class NativeType {
+
+	String refinement = "";
+	NativeTypeEnum type;
+
+	public NativeType(NativeTypeEnum type) {
+		this.type = type;
+	}
+
+	public NativeType(String refinement, NativeTypeEnum type) {
+		this.refinement = refinement;
+		this.type = type;
+	}
+
+	public NativeType() {
```

Comment:
I don't think this constructor makes sense, since we always want a native type enum set. Can we remove it?

Then, from each inheriting class, you can just call super. I include a comment about how to do this in StringType.